application/comit_node/src/http_api/rfc003/handlers/get_swaps.rs

```diff
@@ -73,12 +73,11 @@ pub fn handle_get_swaps<T: MetadataStore<SwapId>, S: StateStore>(
                             parameters,
                         };
 
-                        let mut hal_resource = HalResource::new(swap);
-                        hal_resource.with_link("self", swap_path(id));
+                        let hal_resource = HalResource::new(swap).with_link("self", swap_path(id));
 
                         for action in actions {
                             let link = new_action_link(&id, action);
-                            hal_resource.with_link(action, link);
+                            hal_resource.clone().with_link(action, link);
```

Comment:
:question: why do you need to clone here? Does `with_link` consume?

Looks like you would actually want to `fold` here maybe?

Reply:
That's exactly what I need! Thank you.